Hazel/src/Hazel/Window.h

```diff
@@ -35,13 +33,25 @@ namespace Hazel {
 		virtual unsigned int GetHeight() const = 0;
 
 		// Window attributes
-		virtual void SetEventCallback(const EventCallbackFn& callback) = 0;
+		virtual void SetEventCallback(void (*callback)(Event& e)) { m_Data.EventCallback = callback; };
```

Comment:
This change forces all events to be function pointers, which will not work well for anything that is not a singleton, I believe std::function is going to be the best here

Reply:
Only for the window events though, and window events are handled by the application which is a singleton. I agree though that this isn't optimal.

Reply:
We could maybe look into a delegates library to add to the project. It would be useful on the future. I found this one which claims to be the fastest possible. https://github.com/dreamcat4/FastDelegate

Reply:
Adding a library will require Cherno's approval, and for any sizable library he would probably rather use something he knows. I think adding a library for this is also maybe a bit overkill?

Reply:
Its a single header file

Reply:
I would leave std::function as well... and adding another library should be part of it's own PR?